pkg/namespacestore/validation.go

```diff
@@ -0,0 +1,36 @@
+package namespacestore
+
+import (
+	"fmt"
+	"strings"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+)
+
+// ValidateNamespaceStore validates namespacestore configuration
+func ValidateNamespaceStore(nsStore *nbv1.NamespaceStore) error{
```

Comment:
validateNamespaceStore is a too much general name, you specifically check here validation for namespacestore of type NSFS

Reply:
you are right, there are only NSFS validation for now, but the method is a place to put there all additional namestore validations as well.

Reply:
for other types of namespacestores the validation already exists and taking place inside the function MakeExternalConnectionParams.
I don't find it correct to have only one function in that file, you will probably want to have a switch case in the original function and create multiple validation functions for each type of namespacestore,
but anyway, please move all validations to the file if yo desire to have a validation file for all types